### ISSUE TITLE
Pyright not able to handle __eq__ for Callable

### ISSUE THREAD

**Issue by kumaraditya303** [microsoft/pyright#2417]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

**Describe the bug**
Pyright complains that Callables does not support "== "

**To Reproduce**
Steps to reproduce the behavior.

**Expected behavior**
No error.

**Screenshots or Code**
If applicable, add screenshots or the text of the code (surrounded by triple back ticks) to help explain your problem.
```python
from typing import Callable, Pattern, Union

URLMatch = Union[str, Pattern, Callable[[str], bool]]


def func(a: str) -> None:
    pass

def func2(b: URLMatch) -> None:
    if b == func:
        print("func")
    else: print(type(b))

func2(func)
func2("func")
```

**VS Code extension or command-line**
Are you running pyright as a VS Code extension or a command-line tool? Which version? You can find the version of the VS Code extension by clicking on the Pyright icon in the extensions panel.

```console
pyright 1.1.177
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.178, which I just published. It will also be included in the next release of pylance.
